### PR TITLE
refactor: clean up tf encoder

### PR DESCRIPTION
**Pull request by bwanglzu**
1. more unittest with edge cases
2. refactor integration tests
3. remove `default` from constructor
4. readme polish

### CODE REVIEW

File: jinahub/encoders/image/ImageTFEncoder/README.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
 # ImageTFEncoder
 
-`ImageTFEncoder` encodes ``Document`` content from a ndarray, potentially BatchSize x (Height x Width x Channel) into a ndarray of `BatchSize * d`. Internally, :class:`ImageTFEncoder` wraps the models from `tensorflow.keras.applications`. https://keras.io/applications/
\ No newline at end of file
+`ImageTFEncoder` encodes ``Document`` content from a ndarray, potentially BatchSize x (Height x Width x Channel) into a ndarray of `BatchSize * d`.
+Internally, `ImageTFEncoder` wraps the models from [tensorflow.keras.applications](https://keras.io/applications/).
+
+Use the prebuilt images from JinaHub in your Python code,
```
Comment (cristianmtr) [inline jinahub/encoders/image/ImageTFEncoder/README.md:6]: this part is automatically generated.

File: jinahub/encoders/image/ImageTFEncoder/README.md [hunk 2]
```diff
@@ -1,3 +1,14 @@
...
+
+```python
+from jina import Flow
+
+f = Flow().add(uses='jinahub+docker://ImageTFEncoder',)
+```
```
Comment (cristianmtr) [inline jinahub/encoders/image/ImageTFEncoder/README.md:12]: this

File: jinahub/encoders/image/ImageTFEncoder/README.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
 # ImageTFEncoder
 
-`ImageTFEncoder` encodes ``Document`` content from a ndarray, potentially BatchSize x (Height x Width x Channel) into a ndarray of `BatchSize * d`. Internally, :class:`ImageTFEncoder` wraps the models from `tensorflow.keras.applications`. https://keras.io/applications/
\ No newline at end of file
+`ImageTFEncoder` encodes ``Document`` content from a ndarray, potentially BatchSize x (Height x Width x Channel) into a ndarray of `BatchSize * d`.
```
Comment (nan-wang) [inline jinahub/encoders/image/ImageTFEncoder/README.md:3]:
```suggestion
`ImageTFEncoder` encode the ``blob`` with the size (Height x Width x Channel) of ``Document`` into one dimensional vector and stored it in the ``embedding`` field
```

**Review by nan-wang — CHANGES_REQUESTED** [review]

**Review by nan-wang — APPROVED** [review]

**Review by cristianmtr — APPROVED** [review]
